crates/exex/exex/src/dyn_context.rs

```diff
@@ -0,0 +1,56 @@
+//! Mirrored version of [`ExExContext`](`crate::ExExContext`)
+//! without generic abstraction over [<Node>](`reth_node_api::FullNodeComponents`)
+
+use std::sync::Arc;
+
+use reth_chainspec::{EthChainSpec, Head};
+use reth_node_api::FullNodeComponents;
+use reth_node_core::node_config::NodeConfig;
+use tokio::sync::mpsc;
+
+use crate::{ExExContext, ExExEvent};
+
+// TODO(0xurb) - add `node` and `notifications` after abstractions
+/// Captures the context that an `ExEx` has access to.
+#[derive(Debug)]
+pub struct ExExContextDyn {
```

Comment:
while this works, I wonder if we should instead do a trait abstraction instead so we don't need an extra type for this and instead have a non generic layer on top of ExExContext @shekhirin ?

Reply:
this also works, but some users fully deconstruct the context and then store its fields in their own structs. I think the approach with a mirrored `ExExContext` is better, because it's closer to how a normal ExEx works today.

Reply:
good point, makes sense